os/arch/arm/src/common/up_allocateheap.c

```diff
@@ -209,10 +209,29 @@ void up_allocate_kheap(FAR void **heap_start, size_t *heap_size)
 }
 #endif
 
+/****************************************************************************
+ * Name: up_add_kregion
+ ****************************************************************************/
+#if defined(CONFIG_BUILD_PROTECTED) && (CONFIG_KMM_REGIONS > 1)
```

Comment:
Is it ok when CONFIG_BUILD_KERNEL?
I think it should be not flat build, isn't it?